src/databricks/labs/ucx/hive_metastore/tables.py

```diff
@@ -252,3 +267,24 @@ def _init_seen_tables(self):
 
     def _table_already_upgraded(self, target) -> bool:
         return target in self._seen_tables
+
+    def revert_migrated_tables(self, *, schema: str | None = None, table: str | None = None):
+        def scope_filter(cur_table: Table):
+            schema_match = not schema or cur_table.database == schema
+            # if schema is specified matches the schema
+            table_match = not table or cur_table.name == table
+            # if table is specified matches the table
+            return schema_match and table_match
+
+        upgraded_tables = []
+        for cur_table in list(self._tc.snapshot()):
```

Comment:
Why do you need to wrap snapshot into list()?

Reply:
Fixed